packages/studio-base/src/panels/Map/MapPanel.tsx

```diff
@@ -411,7 +416,7 @@ function MapPanel(props: MapPanelProps): JSX.Element {
       }
 
       if (renderState.allFrames) {
```

Comment:
```suggestion
      if (renderState.allFrames) {
      // use memoized value to avoid unnecessary calls of expensive dependent functions
```
@defunctzombie something like this?

Reply:
```suggestion
      if (renderState.allFrames) {
      // use memoization to avoid re-filtering allFrames when it has not changed
```